src/platform/device.gni

```diff
@@ -56,6 +56,8 @@ declare_args() {
 
 declare_args() {
   chip_openthread_ftd = chip_enable_openthread
+  chip_use_ot_lib = false
```

Comment:
Please document

Please do not abbreviate "openthread". Actually, I do not think this name makes very much sense even if not abbreviated. Both prebuilt and from-source builds are "using" libopenthread.

I don't think device.gni is the right place, it should go in third_party/openthread/BUILD.gn

See third_party/mbedtls/BUILD.gn for an example of a pattern that allows swapping out the library (with anything, including a static library archive). This example is cleaner because the decision is centralized and encapsulates the decision, so it does not pollute other builds files such as src/crypto/BUILD.gn

Reply:
I don't know if it will work in third_party/openthread/BUILD.gn for all dependency due the location. I had some issues with other stuff from third_party/openthread/BUILD.gn being pulled when trying to do it this way.

what about in `src/platform/openthread` . I can create a build.gn there

Reply:
Or i can move all the `third_party/openthread/BUILD.gn` into a gni and use the BUILD.GN for the group. like what is done for mbedtls

Reply:
Should be inline with what you asked now